### PR TITLE
Move examples to an appendix

### PR DESCRIPTION
**Pull request by martinthomson**
I've not modified these much, if at all.  I've added paragraph breaks to
some.

The only one that has changes is TCP, where I've expanded a little on
the discussion of mptcp and tfo, based on my own understanding of the
status of each.  I don't have good citations though, so I'd appreciate
help.

For #53.

### CODE REVIEW

File: draft-iab-use-it-or-lose-it.md [hunk 1]
```diff
@@ -638,6 +552,122 @@ This document makes no request of IANA.
...
+
+When SNMP versions 2, 2c and 3 came along, older agents did exactly what the
+protocol specifies.  Deployment of new versions was likely successful because
+the handling of newer versions was both clear and simple.
+
+
+## TCP
+
+Extension points in TCP {{?TCP=RFC0793}} have been rendered difficult to use,
+largely due to middlebox interactions.
```
Comment (mirjak) [inline draft-iab-use-it-or-lose-it.md]: Reference: Is it Still Possible to Extend TCP?
Michio Honda, Yoshifumi Nishida, Costin Raiciu, Adam Greenhalgh, Mark Handley and Hideyuki Tokuda, ACM Internet Measurement Conference (IMC), November 2011

File: draft-iab-use-it-or-lose-it.md [hunk 2]
```diff
@@ -638,6 +552,122 @@ This document makes no request of IANA.
...
+
+For instance, multipath TCP {{?MPTCP=RFC6824}} can only be deployed
+opportunistically.  As multipath TCP enables progressive enhancement of the
+protocol, this largely only causes the feature to not be available if the path
+is intolerant of the extension.
```
Comment (mirjak) [inline draft-iab-use-it-or-lose-it.md]: MPTCP specific reference: How Hard Can It Be? Designing and Implementing a Deployable Multipath TCP
Costin Raiciu, Christoph Paasch, Sebastien Barre, Alan Ford, Michio Honda, Fabien Duchene, Olivier Bonaventure and Mark Handley, USENIX Networked Systems Design and Implementation (NSDI), April 2012

File: draft-iab-use-it-or-lose-it.md [hunk 3]
```diff
@@ -303,14 +224,7 @@ to full participation.  For example, a SIP back-to-back user agent (B2BUA)
...
 This phenomenon appears at all layers of the protocol stack, especially when
-protocols are not designed with middlebox participation in mind. TCP's
-{{?TCP=RFC0793}} extension points have been rendered difficult to use, largely
-due to middlebox interactions; see for example Multipath TCP {{?MPTCP=RFC6824}}
-or Fast Open {{?TFO=RFC7413}}.  IP's version field was rendered useless when
-encapsulated over Ethernet, requring a new ethertype with IPv6 {{?RFC2464}}, due
-in part to layer 2 devices making version-independent assumptions about the
-structure of the IPv4 header.  The announcements of new optional transitive
-attributes in BGP caused significant routing instability {{RIPE-99}}.
+protocols are not designed with middlebox participation in mind.
```
Comment (mirjak) [inline draft-iab-use-it-or-lose-it.md]: Add a pointer to the new appendix...?

**Comment by martinthomson** [comment]
Thanks Mirja, those are great references.

**Comment by martinthomson** [comment]
@tfpauly, what do you think?

**Comment by tfpauly** [comment]
@martinthomson looks good to me! I think Section 2.2 now looks a little short in context, but that's probably OK.

**Review by tfpauly — APPROVED** [review]

**Pull request merged**